File: training-data/elixir/max_subarray.exs
defmodule MaxSubarray do
  def kadane(list) do
    [head | tail] = list

    {best, _current} =
      Enum.reduce(tail, {head, head}, fn x, {best, current} ->
        current = max(x, current + x)
        {max(best, current), current}
      end)

    best
  end
end

IO.inspect(MaxSubarray.kadane([-2, 1, -3, 4, -1, 2, 1, -5, 4]))

File: training-data/erlang/max_subarray.erl
-module(max_subarray).
-export([kadane/1]).

kadane([H | T]) ->
    {Best, _Current} = lists:foldl(
        fun(X, {Best, Current}) ->
            NewCurrent = max(X, Current + X),
            {max(Best, NewCurrent), NewCurrent}
        end,
        {H, H},
        T
    ),
    Best.
